### PR TITLE
adding the option to download the Dataset File

### PR DESCRIPTION
**Pull request by AdarshdeepCheema**
Signed-off-by: Adarshdeep Cheema <adarshdeep.cheema@ibm.com>

### CODE REVIEW

File: src/app/services/downloader.service.ts [hunk 1]
```diff
@@ -69,8 +69,23 @@ export class DownloaderService {
...
+            if(downloadObject.data.isDataset) {
+              response.json().then(json => {
+                reader = json.records.filter(function(record){return record.length > 0}).map(function(record){return record.trim()}).join("\n");
+                const blob = new Blob([reader], { type: 'text/plain' });
+                createAndDownloadElement(blob, downloadObject.data.path);
+                resolve();
+              })
+              .catch(error => {
+                context.log.severe("An error occurred downloading " + fileName + " : ", error)
+                controller.error(error);
```
Comment (DivergentEuropeans) [inline src/app/services/downloader.service.ts:82]: I think ReadableStream may print its own console error when you declare error like this (so you could probably remove error object from log or vice versa so we don't have duplicates)

Comment (DivergentEuropeans) [reply src/app/services/downloader.service.ts:82]: Yes, it does. Tested by shutting and restarting ZSS to cause error

**Review by DivergentEuropeans — APPROVED** [review]
This looks gtm! Please see comment & make second PR into v2.x/staging

File: src/app/services/downloader.service.ts (hunk 1 above)
Comment (AdarshdeepCheema) [reply src/app/services/downloader.service.ts:82]: done

**Pull request merged**